guacamole/src/main/webapp/app/clipboard/directives/guacClipboard.js

```diff
@@ -0,0 +1,312 @@
+     */
+    var config = {
+        restrict    : 'E',
+        replace     : true,
+        templateUrl : 'app/clipboard/templates/guacClipboard.html'
+    };
+
+    // Scope properties exposed by the guacClipboard directive
+    config.scope = {
+
+        /**
+         * The data to display within the field provided by this directive. If
+         * the local clipboard can be accessed by JavaScript, this will be set
+         * automatically as the local clipboard changes. Failing that, this
+         * will be set when the user manually modifies the contents of the
+         * field. Changes to this value will be rendered within the field and,
+         * if possible, will be pushed to the local clipboard.
+         *
+         * @type ClipboardData
+         */
+        data : '='
+
+    };
+
+    // guacClipboard directive controller
+    config.controller = ['$scope', '$injector', '$element',
+            function guacClipboardController($scope, $injector, $element) {
+
+        // Required services
+        var $rootScope       = $injector.get('$rootScope');
+        var $window          = $injector.get('$window');
+        var clipboardService = $injector.get('clipboardService');
+
+        /**
+         * Reference to the window.document object.
+         *
+         * @private
+         * @type HTMLDocument
+         */
+        var document = $window.document;
+
+        /**
+         * Map of all currently pressed keys by keysym. If a particular key is
+         * currently pressed, the value stored under that key's keysym within
+         * this map will be true. All keys not currently pressed will not have entries
+         * within this map.
+         *
+         * @type Object.<Number, Boolean>
+         */
+        var keysCurrentlyPressed = {};
+
+        /**
+         * Map of all currently pressed keys (by keysym) to the clipboard
+         * contents received while those keys were pressed. All keys not
+         * currently pressed will not have entries within this map.
+         *
+         * @type Object.<Number, Blob>
+         */
+        var clipboardDataFromKey = {};
+
+        /**
+         * The FileReader to use to read File or Blob data received from the
+         * clipboard.
+         *
+         * @type FileReader
+         */
+        var reader = new FileReader();
+
+        /**
+         * The content-editable DOM element which will contain the clipboard
+         * contents within the user interface provided by this directive.
+         *
+         * @type Element
+         */
+        var element = $element[0];
+
+        // Intercept paste events, handling image data specifically
+        element.addEventListener('paste', function dataPasted(e) {
+
+            // Always clear the current clipboard contents upon paste
+            element.innerHTML = '';
+
+            // If we can't read the clipboard contents at all, abort
+            var clipboardData = e.clipboardData;
```

Comment:
If there's a way to decide whether it's an image before clearing the text, I think that would be good to do. It's unexpected and undesirable in my opinion for all the text currently in the clipboard to disappear without warning if you paste more text. I think it will make people unhappy.